### PR TITLE
new event card design

### LINKED ISSUE #162: new event card design
https://www.figma.com/file/WCc4V3sy05iF55BRum71Mv/Portal-UI-v2-Pages?type=design&node-id=1821-3350&mode=design&t=wDmcRKJqxpOdm7nl-0

image

beware of conflicts from #159, it makes changes to the component

### PR DESCRIPTION
**Pull request by SheepTester**
# Info

Closes #162 

updated event card design

did not touch event details modal, so `PointsDisplay` and `CommunityLogo` are still in use

## Changes

- event card design
- I changed the `Community.AI` enum value to `Ai`. The enum was only used for `CommunityLogo`. There are two reasons for this
  - `CommunityLogo` currently uses `capitalize` to normalize ACM communities, but this would turn `AI` into `Ai`
  - For some reason, our `.scss.d.ts` generator turns `.badgeAI` into `.badgeAi`. I have no idea why

# Type of Change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) 

# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally on Desktop.
- [x] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [x] My changes produce no new warnings.

# Screenshots

image

image

### CODE REVIEW

**Review by alexzhang1618 — CHANGES_REQUESTED** [review]
Contrast is a little rough on dark mode
Screenshot 2024-03-11 at 8 17 54 PM

**Comment by alexzhang1618** [comment]
> Contrast is a little rough on dark mode

Also, is it just me or is the spacing a little uneven between the event title, event location, and tags? Let's also remove the year when on the main dashboard as well since it should be evident that it's happening this year

**Comment by SheepTester** [comment]
> Also, is it just me or is the spacing a little uneven between the event title, event location, and tags? Let's also remove the year when on the main dashboard as well since it should be evident that it's happening this year

the year is there because the event is a year long

**Comment by SheepTester** [comment]
> [whatever alex said about contrast]

yeah i realized it has been fixed

**Review by alexzhang1618 — APPROVED** [review]
lgtm ship ship ship ship ship ship ship ship ship ship ship ship ship ship ship ship ship ship ship ship

**Comment by SheepTester** [comment]
issue: pickup event on checkout page is cut off

File: src/components/common/CommunityLogo/index.tsx [hunk 1]
```diff
@@ -9,14 +8,8 @@ interface CommunityLogoProps {
 }
 
 const CommunityLogo = ({ community, size }: CommunityLogoProps) => {
-  const formattedName = capitalize(community) as Community;
-
-  if (!Object.values(Community).includes(formattedName))
-    return <Image src={communityLogos.General} width={size} alt="ACM General Logo" />;
```
Comment (farisashai) [inline src/components/common/CommunityLogo/index.tsx:15]: is this completely removing the fallback for an invalid enum value?

File: src/components/events/EventCard/index.tsx [hunk 2]
```diff
@@ -75,29 +78,43 @@ const EventCard = ({
...
+                style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}
+              >
+                {location}
+              </Typography>
+            </div>
+            <div className={styles.badges}>
+              <div className={`${styles.badge} ${styles[`badge${community}`]}`}>
+                {communityNames[community]}
               </div>
+              {!isOrderPickupEvent(event) && (
```
Comment (farisashai) [inline src/components/events/EventCard/index.tsx]: left a comment abt this syntax on a older PR. we should prefer `condition ? Component : null`

File: src/components/common/CommunityLogo/index.tsx (hunk 1 above)
Comment (SheepTester) [reply src/components/common/CommunityLogo/index.tsx:15]: yes, because the fallback defaults to ACM General

**Comment by SheepTester** [comment]
It shouldn't show any tags for unlinked events

image

**Pull request merged**